66_HLSLBxDFTests/app_resources/tests.hlsl

```diff
@@ -0,0 +1,455 @@
+#ifndef BXDFTESTS_TESTS_HLSL
+#define BXDFTESTS_TESTS_HLSL
+
+#include "nbl/builtin/hlsl/cpp_compat.hlsl"
+
+#include "nbl/builtin/hlsl/random/xoroshiro.hlsl"
+#include "nbl/builtin/hlsl/bxdf/common.hlsl"
+#include "nbl/builtin/hlsl/bxdf/reflection.hlsl"
+#include "nbl/builtin/hlsl/bxdf/transmission.hlsl"
+
+namespace nbl
+{
+namespace hlsl
+{
+
+using ray_dir_info_t = bxdf::ray_dir_info::SBasic<float>;
+using iso_interaction = bxdf::surface_interactions::SIsotropic<ray_dir_info_t>;
+using aniso_interaction = bxdf::surface_interactions::SAnisotropic<ray_dir_info_t>;
+using sample_t = bxdf::SLightSample<ray_dir_info_t>;
+using iso_cache = bxdf::SIsotropicMicrofacetCache<float>;
+using aniso_cache = bxdf::SAnisotropicMicrofacetCache<float>;
+using quotient_pdf_t = bxdf::quotient_and_pdf<float32_t3, float>;
+
+inline float rngFloat01(NBL_REF_ARG(nbl::hlsl::Xoroshiro64Star) rng)
+{
+    return (float)rng() / numeric_limits<uint32_t>::max;
+}
+
+inline float32_t3 rngFloat301(NBL_REF_ARG(nbl::hlsl::Xoroshiro64Star) rng)
+{
+    return float32_t3(rngFloat01(rng), rngFloat01(rng), rngFloat01(rng));
+}
+
+
+struct SBxDFTestResources
+{
+    static SBxDFTestResources create(uint32_t2 seed)
+    {
+        SBxDFTestResources retval;
+
+        retval.rng = nbl::hlsl::Xoroshiro64Star::construct(seed);
+        retval.u = float32_t3(rngFloat01(retval.rng), rngFloat01(retval.rng), 0.0);
+
+        retval.V.direction = nbl::hlsl::normalize<float32_t3>(rngFloat301(retval.rng));
+        retval.N = nbl::hlsl::normalize<float32_t3>(rngFloat301(retval.rng));
```

Comment:
directions should be generated differently, there's a [0,1]^2 -> Hemisphere / Sphere transform code you should test first and then use throughout